tools/crm_mon.c

```diff
@@ -840,10 +841,21 @@ main(int argc, char **argv)
 
         if (rc != pcmk_ok) {
             if (output_format == mon_output_monitor) {
-                printf("CLUSTER WARN: Connection to cluster failed: %s\n", pcmk_strerror(rc));
-                clean_up(MON_STATUS_WARN);
+                if (rc == -ENOTCONN) {
+                    printf("CLUSTER ERROR: cluster is not available on this node\n");
+                    clean_up(MON_STATUS_ERR);
+                } else {
+                    printf("CLUSTER WARN: Connection to cluster failed: %s\n",
+                           pcmk_strerror(rc));
+                    clean_up(MON_STATUS_WARN);
+                }
```

Comment:
Looking at this code again, I'm thinking monitor output doesn't need to distinguish -ENOTCONN from other errors, and the return value should always be CRIT, as it indicates a serious cluster problem. A case could be made for unknown (3 / "CLUSTER UNKNOWN"), which is intended for cases where a definite good or bad answer could not be obtained, but I think in a cluster context, not being able to connect to the cluster is worthy of a red alarm.

Reply:
Agreed about the return value to be always CRIT.
Just had to create a case to force the error-message as agreed on in the bz.

Reply:
or did you just mean for the nagios-case?